### PR TITLE
KAFKA-17073: Deprecate ReplicaVerificationTool in 3.9

### PR DESCRIPTION
**Pull request by dongjinleekr**
### Summary

If the user runs `ReplicaVerificationTool`, it shows a deprecation warning.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: tools/src/main/java/org/apache/kafka/tools/ReplicaVerificationTool.java [hunk 1]
```diff
@@ -112,6 +112,8 @@ public class ReplicaVerificationTool {
 
     public static void main(String[] args) {
         try {
+            LOG.warn("This tool is deprecated and may be removed in a future major release.");
```
Comment (chia7712) [inline tools/src/main/java/org/apache/kafka/tools/ReplicaVerificationTool.java:115]: Could you please add "3.9.0" to this message?

**Review by chia7712 — COMMENTED** [review]
@dongjinleekr thanks for this patch

File: tools/src/main/java/org/apache/kafka/tools/ReplicaVerificationTool.java (hunk 1 above)
Comment (dongjinleekr) [reply tools/src/main/java/org/apache/kafka/tools/ReplicaVerificationTool.java:115]: @chia7712

I followed the consistency with `kafka.tools.MirrorMaker` and `kafka.admin.FeatureCommand`.

`kafka.tools.MirrorMaker`:

```
bin/kafka-mirror-maker.sh --help

[2024-07-15 20:08:47,486] WARN This tool is deprecated and may be removed in a future major release. (kafka.tools.MirrorMaker$)
This tool helps to continuously copy data between two Kafka clusters.
...
```

`kafka.admin.FeatureCommand`

```
bin/kafka-run-class.sh kafka.admin.FeatureCommand --help

WARNING: The 'kafka.tools' package is deprecated and will change to 'org.apache.kafka.tools' in the next major release.
usage: kafka-features [-h] [--command-config COMMAND_CONFIG] (--bootstrap-server BOOTSTRAP_SERVER | --bootstrap-controller BOOTSTRAP_CONTROLLER) {describe,upgrade,downgrade,disable} ...

This tool manages feature flags in Kafka.
...
```

If we really need the version, how about this? Is this okay?

> This tool is deprecated <ins>from 3.9.0</ins> and may be removed in a future major release.

Comment (chia7712) [reply tools/src/main/java/org/apache/kafka/tools/ReplicaVerificationTool.java:115]: I'd like to add deprecated version and removed version if it does not have replacement.

However, that is sort of personally taste. I'm fine to keep the origin message for consistency 😄

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request merged**